test/common/user_permissions.go

```diff
@@ -0,0 +1,227 @@
+	rhmi := &v1alpha1.RHMI{}
+	ns := fmt.Sprintf("%soperator", namespacePrefix)
+	if err := ctx.Client.Get(goctx.TODO(), types.NamespacedName{Name: installationName, Namespace: ns}, rhmi); err != nil {
+		t.Fatal("error getting RHMI CR:", err)
+	}
+	// get console master url
+	masterURL := rhmi.Spec.MasterURL
+
+	// get oauth route
+	oauthRoute := &v1.Route{}
+	if err := ctx.Client.Get(goctx.TODO(), types.NamespacedName{Name: openshiftOAuthRouteName, Namespace: openshiftAuthenticationNamespace}, oauthRoute); err != nil {
+		t.Fatal("error getting Openshift Oauth Route: ", err)
+	}
+
+	// get rhmi developer user tokens
+	rhmiDevUserToken, err := resources.GetUserOpenshiftToken(oauthRoute.Spec.Host, masterURL, resources.DefaultIDP, "test-user01", "Password1")
+	if err != nil {
+		t.Fatalf("error occured trying to get token : %v", err)
+	}
+
+	// get projects for rhmi developer
+	rhmiDevfoundProjects, err := ocGetProjects(masterURL, rhmiDevUserToken)
+	if err != nil {
+		t.Fatalf("error occured while getting user projects : %v", err)
+	}
+
+	// check if projects are as expected for rhmi developer
+	for projectCount, p := range rhmiDevfoundProjects.Items {
+		t.Log(fmt.Sprintf("found rhmi-developer project - %s", p.Name))
+		if projectCount >= expectedRhmiDeveloperProjectCount {
+			t.Fatal(fmt.Sprintf("test failed - project count for rhmi-developer exceeded expected 1"))
+		}
+		if p.Name != expectedRhmiDeveloperNamespace {
+			t.Fatal(fmt.Sprintf("test failed - found project for rhmi-developer does not match expected : %s", expectedRhmiDeveloperNamespace))
+		}
+	}
+	t.Log("test-passed - found projects for rhmi-developer are as expected")
+
+	// get dedicated admin token
+	dedicatedAdminToken, err := resources.GetUserOpenshiftToken(oauthRoute.Spec.Host, masterURL, resources.DefaultIDP, "customer-admin01", "Password1")
+	if err != nil {
+		t.Fatalf("error occured trying to get token : %v", err)
+	}
+
+	// get projects for dedicated admin
+	dedicatedAdminFoundProjects, err := ocGetProjects(masterURL, dedicatedAdminToken)
+	if err != nil {
+		t.Fatalf("error occured while getting user projects : %v", err)
+	}
+
+	// check if projects are as expected for rhmi developer
+	if result := verifyFoundProjects(dedicatedAdminFoundProjects.Items); !result {
+		t.Fatal("test-failed - projects missing for dedicated-admins")
+	}
+	t.Log("test-passed - found projects for dedicated-admins are as expected")
+
+	// get fuse pods for rhmi developer
+	podlist, err := ocGetNameSpacePods(masterURL, pathFusePods, rhmiDevUserToken)
+	if err != nil {
+		t.Fatalf("error occured while getting pods : %v", err)
+	}
+
+	// check if six pods are running
+	runningCount := 0
+	for _, p := range podlist.Items {
+		if p.Status.Phase == "Running" {
+			runningCount++
+		}
+	}
+	if runningCount != expectedFusePodCount {
+		t.Fatalf("test-failed - %d found fuse pod count not match expected count of %d", runningCount, expectedFusePodCount)
```

Comment:
Could this error message be slightly clearer? Something like: `expected fuse pod count: <count>, found fuse pod count: <count>`